wpas-hl7-translator/src/main/java/wales/nhs/dhcw/inthub/wpasHl7/MprToAdtA40Mapper.java

```diff
@@ -0,0 +1,172 @@
+package wales.nhs.dhcw.inthub.wpasHl7;
+
+import ca.uhn.hl7v2.model.DataTypeException;
+import ca.uhn.hl7v2.model.v251.message.ADT_A39;
+import wales.nhs.dhcw.inthub.wpasHl7.xml.MAINDATA;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class MprToAdtA40Mapper {
+    private ADT_A39 adtMessage;
+    private final DateTimeProvider dateTimeProvider;
+
+    public MprToAdtA40Mapper(DateTimeProvider dateTimeProvider) {
+        this.dateTimeProvider = dateTimeProvider;
+        adtMessage = new ADT_A39();
+    }
+
+    public ADT_A39 ADT_A40Mapper(MAINDATA.TRANSACTION transaction) throws DataTypeException {
+        if (MshMapper(transaction) != null) {
+            if (EVNMapper(transaction) != null && PIDMapper(transaction) != null && PD1Mapper(transaction) != null) {
+                MRGMapper(transaction);
+            }
+        } else {
+            throw new RuntimeException("Not Able to Map XLM to ADT_A40");
+        }
+        return adtMessage;
+    }
+
+
+    public ADT_A39 MshMapper(MAINDATA.TRANSACTION transaction) throws DataTypeException {
+
+        try {
+            // Mapping header
+            adtMessage.getMSH().getFieldSeparator().setValue("|");
+            adtMessage.getMSH().getEncodingCharacters().setValue("^~\\&");
```

Comment:
it's a good practice to define hardcoded strings as contants (especially that they would need to be reused in additional messages)